include/gpioMonitor.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <sdbusplus/asio/connection.hpp>
+
+namespace vpd
+{
+/**
+ * @brief class for GPIO event handling.
+ *
+ * Responsible for detecting events and handling them. It continuously
+ * monitors the presence of the FRU. If any attachment or detachment is
+ * detected, it enables or disables the FRU's output GPIO and binds or
+ * unbinds the driver accordingly.
+ */
+class GpioEventHandler
+{
+  public:
+    GpioEventHandler() = delete;
+    ~GpioEventHandler() = delete;
+    GpioEventHandler(const GpioEventHandler&) = delete;
+    GpioEventHandler& operator=(const GpioEventHandler&) = delete;
+    GpioEventHandler(GpioEventHandler&&) = delete;
+    GpioEventHandler& operator=(GpioEventHandler&&) = delete;
+
+    /**
+     * @brief Constructor
+     *
+     * @param[in] i_presencePin - GPIO presence pin
+     * @param[in] i_presenceValue - holds the value of presence pin
+     * @param[in] i_outputPin - GPIO output pin
+     * @param[in] i_outputValue - value of the output pin
+     * @param[in] i_inventoryPath - inventory path of the FRU.
+     * @param[in] i_ioContext - pointer to the io context object
+     */
+    GpioEventHandler(std::string& i_presencePin, bool& i_presenceValue,
+                     std::string& i_outputPin, bool& i_outputValue,
+                     std::string& i_inventoryPath,
+                     std::shared_ptr<boost::asio::io_context>& i_ioContext) :
+        m_presencePin(i_presencePin),
```

Comment:
would it be better to keep It as input pin instead of presence pin

Reply:
Updated